modules/publicize/class-async-publicize.php

```diff
@@ -0,0 +1,343 @@
+	 *
+	 * @param WP_REST_Request $request Request instance from REST call.
+	 * @return string|null Result body to return to REST caller
+	 */
+	public function publicize_post( $request ) {
+		$post_id = $request['post_id'];
+		// Message to be used when sharing to social networks.
+		$message = $request->get_header( 'message' );
+
+		if ( ! $this->can_publicize_post( $post_id ) ) {
+			/*
+			 * This is not necessarily an error. The post was filtered by the plugin
+			 * or some hook that was added in. Just return null.
+			 */
+			return null;
+		}
+
+		// Get global id of blog on WPCOM.
+		$blog_id       = Jetpack_Options::get_option( 'id' );
+		$endpoint_url  = self::API_BASE_URL . self::API_VERSION . '/sites/' . $blog_id . '/posts/' . $post_id . '/publicize';
+		$endpoint_url .= '?message=' . sanitize_text_field( $message );
+
+		$request_args = array(
+			'url'     => $endpoint_url,
+			'method'  => 'POST',
+			'timeout' => self::WPCOM_REST_TIMEOUT,
+			'user_id' => JETPACK_MASTER_USER,
+		);
+
+		$result = Jetpack_Client::remote_request( $request_args );
+
+		// Pass through body of response (contains list of succesful connection post ids).
+		return $result['body'];
+	}
+
+	/**
+	 * Checks if post should be filtered out
+	 *
+	 * Calls back to filter method of main Publicize class
+	 * to disable publicize on the post if necessary.
+	 *
+	 * @since 5.9.1
+	 * @see publicize-jetpack.php/set_post_flags()
+	 *
+	 * @param  int $post_id ID number of post.
+	 * @return boolean True if post can be publicized, false otherwise.
+	 */
+	public function can_publicize_post( $post_id ) {
+		$empty_flags     = array();
+		$post            = get_post( $post_id );
+		$publicize_flags = $this->publicize->set_post_flags( $empty_flags, $post );
+
+		if ( isset( $publicize_flags['publicize_post'] ) && $publicize_flags['publicize_post'] ) {
+			return true;
+		} else {
+			return false;
+		}
+	}
+
+	/**
+	 * Completes any necessary cleanup after a post is published.
+	 *
+	 * Does cleanup after Jetpack is finished sending post for publish.
+	 * Useful for cleaning up the temporary meta key that blocks publicize.
+	 *
+	 * @since 5.9.1
+	 *
+	 * @param int   $post_id ID number of post being published.
+	 * @param array $flags {@see class.jetpack-sync-module-posts.php/send_published()}.
+	 */
+	public function post_publish_cleanup( $post_id, $flags ) {
+		// Clean post meta since it's purpose has been served.
+		delete_post_meta( $post_id, self::PUBLICIZE_BLOCK_META_KEY );
+	}
+
+	/**
+	 * Enqueue scripts when they are needed for the edit page
+	 *
+	 * Enqueues necessary scripts for edit page.
+	 * is open.
+	 *
+	 * @since 5.9.1
+	 *
+	 * @param string $hook_suffix File name of current page.
+	 */
+	public function post_page_enqueue( $hook_suffix ) {
+		if ( ( 'post.php' === $hook_suffix || 'post-new.php' === $hook_suffix ) ) {
+			wp_enqueue_script( 'async_publicize_js', plugins_url( 'assets/async-publicize.js', __FILE__ ), array( 'jquery' ) );
```

Comment:
You may want to add a dependency for the `wp-blocks` and `wp-element` script handles as well:
https://github.com/WordPress/gutenberg/tree/v2.4.0/blocks#getting-started